### DISCUSSION TITLE
如何添加新的special token，诸如<user>,<assistant>这种token

### DISCUSSION THREAD

**Discussion by minmie** [zai-org/ChatGLM3#1230] [Q&A] [answered]
```如<user>,<assistant>这些special token，如果我想添加一个<thought>的special token，我应该如何添加？ ```

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
在这改
https://huggingface.co/THUDM/chatglm3-6b/blob/main/tokenizer_config.json
但是这没办法增加一个全新的 6开头的 比如 64790这种token，因为这设计到重新训练词表

**Reply by minmie** [reply]
这个config文件应该不是手动编写的吧，应该是生成的吧，我猜测是不是应该在训练的时候添加新token，训练完毕后，生成的这个配置文件里就有这个token了。但是具体的不知如何操作
